### PR TITLE
Degree status updated

### PR DESCRIPTION
**Pull request by goodviber**
### JIRA ticket number
WIP DO NOT APPROVE/MERGE
### Context
we need to match candidate submission data for the different pathways to api validation constraints
### Changes proposed in this pull request
Candidate Submission service and candidate module
Updating form attributes to match api fields
Session var for privacy policy
### Guidance to review
Error handling to be returned to when api client updated.
Refactoring of integer attributes

### CODE REVIEW

File: app/helpers/session_helper.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def show_session(question)
   end
 
   def show_link(step)
-    "<a href='#{new_registration_path(step)}'>Change</a>".html_safe
+    "<a href='#{new_registration_path(step)}' class='link-color'>Change</a>".html_safe
```
Comment (ethax-ross) [inline app/helpers/session_helper.rb:8]: Did you run this past Ben?

File: app/models/have_a_degree.rb [hunk 2]
```diff
@@ -1,31 +1,46 @@
 class HaveADegree < Base
+  attribute :degree_options, :string
   attribute :degree_status_id, :string
   attribute :degree_type_id, :string
-
+# change this again d
```
Comment (ethax-ross) [inline app/models/have_a_degree.rb:5]: Typo - is this still needed?

File: spec/models/concerns/candidate_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require "rails_helper"
+
+RSpec.describe Candidate do
+  it "has a RETURNER constant" do
```
Comment (ethax-ross) [inline spec/models/concerns/candidate_spec.rb:4]: Were we not dropping these?

File: spec/requests/pages_spec.rb [hunk 4]
```diff
@@ -30,5 +30,24 @@
...
     end
+
+    context "viewing a specific privacy policy" do
+      subject do
+        get privacy_policy_path(id: policy.id)
+        response
+      end
+
+      before do
+        allow_any_instance_of(GetIntoTeachingApiClient::PrivacyPoliciesApi).to receive(:get_privacy_policy).with(policy.id).and_return(policy)
```
Comment (ethax-ross) [inline spec/requests/pages_spec.rb:41]: Can we have a test here to ensure the session is set?

File: app/services/candidate_submission.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    end
+  end
+
+  def filter_candidate(path)
+    data = @session[:registration].select { |key, _| path.include? key.to_sym }
+    merge_integers(data)
+  end
+
+  def merge_integers(body)
+    str_values = %w[
```
Comment (ethax-ross) [inline app/services/candidate_submission.rb:37]: I think this should be fixed before this PR is merged down. In my mind it should just be changing these attributes to type `:integer`:

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/e74328ddae494c9580ec7d22e7bb4ef37807649e/app/models/what_degree_class.rb#L4

And then `to_s` the value here:

https://github.com/DFE-Digital/get-teacher-training-adviser-service/blob/master/app/validators/types_validator.rb#L5

File: spec/models/concerns/candidate_spec.rb (hunk 3 above)
Comment (goodviber) [reply spec/models/concerns/candidate_spec.rb]: 👍

File: spec/requests/pages_spec.rb (hunk 4 above)
Comment (goodviber) [reply spec/requests/pages_spec.rb:28]: 👍

File: app/helpers/session_helper.rb (hunk 1 above)
Comment (goodviber) [reply app/helpers/session_helper.rb:8]: Yes, figured out the issue, put it on another ticket.

File: app/models/have_a_degree.rb (hunk 2 above)
Comment (goodviber) [reply app/models/have_a_degree.rb]: Removed

File: app/services/candidate_submission.rb (hunk 5 above)
Comment (goodviber) [reply app/services/candidate_submission.rb:37]: Not opposed to doing this, but it needs to be a sep PR, it touches a lot of files, and the verification process to check the correct data is getting posted is prolonged, I have opened a ticket.

**Comment by jebw** [comment]
Is this still relevant or can it be closed?

**Pull request closed without merging**